pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -495,6 +497,14 @@ public boolean isIndexUseAllowed(String columnName, FieldConfig.IndexType indexT
     return !_skipIndexes.getOrDefault(columnName, Collections.EMPTY_SET).contains(indexType);
   }
 
+  public void setHasSchemaMisMatch(boolean hasSchemaMismatch) {
```

Comment:
Why are these setters/getters needed in QueryContext? We are detecting this in BrokerReduceService right?